### ISSUE TITLE
[Feature]: Allow disabling rows in a multi-select table

### ISSUE THREAD

**Issue by kamal-buqaileh** [appsmithorg/appsmith#37350]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

We have a use case where we want to display some data to our users in a table and then allow the user to select some table records to perform some calculations.

Some of the records are not suitable for the calculation. However, we still want the user to be able to view these records, but without the ability to select a record.

basically, we want the ability to disable some records in a multi-select table so that the user can't select the records

### Why should this be worked on?

it is a nice addition to your software, also I assumed it exists by default in Appsmith
